meta/store.go

```diff
@@ -1624,10 +1636,23 @@ type storeFSMSnapshot struct {
 }
 
 func (s *storeFSMSnapshot) Persist(sink raft.SnapshotSink) error {
-	// TODO: Encode data.
-	// TODO: sink.Write(p)
-	// TODO: sink.Close()
-	panic("not implemented yet")
+	// Encode data.
+	p, err := s.Data.MarshalBinary()
+	if err != nil {
+		return err
+	}
+
+	// Write data to sink.
+	if _, err := sink.Write(p); err != nil {
+		return err
+	}
+
+	// Close the sink.
+	if err := sink.Close(); err != nil {
+		return err
+	}
```

Comment:
`sink.Cancel()` needs to be called if any error occurs. It's probably easiest if you wrap the whole thing in a `func() error` closure and check the return and call `Cancel()` if there's an error.